src/perftest_resources.c

```diff
@@ -3304,6 +3312,9 @@ void ctx_set_send_reg_wqes(struct pingpong_context *ctx,
 				(user_param->connection_type == RawEth) ? (user_param->size - HW_CRC_ADDITION) : user_param->size;
 
 			ctx->sge_list[i*user_param->post_list + j].lkey = ctx->mr[i]->lkey;
+			if (user_param->use_null_mr && (user_param->verb == WRITE || user_param->verb == READ)) {
```

Comment:
after adding the restriction in the perftest_parameters.c, you can remove the check of the verb type from here.

Reply:
Good catch, thanks, updated.